src/LibraryViewExtensionWebView2/LibraryViewController.cs

```diff
@@ -753,5 +753,13 @@ public static async Task<string> ExecuteScriptFunctionAsync(WebView2 webView2, s
             script += ");";
             return await webView2.ExecuteScriptAsync(script);
         }
+        /// <summary>
+        /// Gives the library UI information on Dynamo's current context.
+        /// </summary>
+        /// <param name="type"></param>
+        internal void UpdateContext(string type)
+        {
+            ExecuteScriptFunctionAsync(browser,"libController.setHostContext", type);
```

Comment:
Usually we were defining the function in the library.html (like an interface of Dynamo and librarie.js), for example:
function SetContext(type){
     libController.setHostContext(type)
}
But I guess this way is better if now everything will be inside the librarie.js component.

Reply:
I actually had trouble with that pattern where `libController` was not set correctly.

Reply:
I don't think that we're using ExecuteScriptFunctionAsync for so many interops between c# and javascript. ExecuteScriptFunctionAsync forces you to write the javascript code inside c# (prone to errors)
There are other ways to interop (post simple  messages or hostObjects) We should review the interop approach in a separate task